clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -315,6 +315,19 @@ private synchronized void disableHeartbeatThread() {
             heartbeatThread.disable();
     }
 
+    private void closeHeartbeatThread() {
+        if (heartbeatThread != null) {
+            heartbeatThread.close();
+
+            try {
+                heartbeatThread.join();
+            } catch (InterruptedException e) {
+                log.error("Interrupted while waiting for consumer heartbeat thread to close");
```

Comment:
Should this be a warning or even `info`? It doesn't seem like there's much for the user to do in this case.